tests/test_platform_util.py

```
@@ -1603,3 +1698,793 @@ def test_walk_files_unlinked_refuses_a_link_like_top(tmp_path, monkeypatch):
+    `_retry_on_sharing_violation`'s bounded backoff, and cleans its temp up.
+
+    Ablation: flip `require_writable_target`'s default to True and this reddens
+    on `calls` staying empty — `_refuse_unwritable_target` raises before
+    `_mkstemp_beside` runs, so nothing stages."""
+    target = tmp_path / "sprint-status.yaml"
+    target.write_text("before", encoding="utf-8")
+    target.chmod(0o444)
+    calls: list[str] = []
+    real_mkstemp = tempfile.mkstemp
+
+    def spy(*, dir, prefix, suffix):
```

Ruff (0.16.2)

[error] Function argument `dir` is shadowing a Python builtin

(A002)